Org.onionshare.onion share

micahflee opened the pull request:
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

kushaldas commented:
It will be an amazing addition. Thank you @micahflee.

barthalion commented on README.md:
```diff
@@ -1,27 +1,8 @@
-# Flathub
```
Feel free to delete this file too.

barthalion commented on org.onionshare.OnionShare.yaml:
```diff
@@ -0,0 +1,336 @@
+  - "--share=ipc"
+  - "--share=network"
+  - "--socket=wayland"
+  - "--socket=x11"
```
```suggestion
  - "--socket=fallback-x11"
```

barthalion commented on org.onionshare.OnionShare.yaml:
```diff
@@ -0,0 +1,336 @@
+  - "--talk-name=org.freedesktop.secrets"
+  - "--filesystem=home:ro"
+  - "--filesystem=~/OnionShare:create"
+  - "--filesystem=xdg-config/onionshare:create"
```
```suggestion
```

Your application has write access to `$XDG_CONFIG_HOME` by default.

barthalion commented on org.onionshare.OnionShare.yaml:
```diff
@@ -0,0 +1,336 @@
+  - "--share=network"
+  - "--socket=wayland"
+  - "--socket=x11"
+  - "--talk-name=org.freedesktop.Flatpak"
```
What do you need it for?

barthalion commented on org.onionshare.OnionShare.yaml:
```diff
@@ -0,0 +1,336 @@
+          - mkdir -p /app/include/qt5tmp && cp -R /usr/include/Qt* /app/include/qt5tmp # https://bugreports.qt.io/browse/PYSIDE-787
+          - sed -i 's|\(--include-paths=\)|\1/app/include/qt5tmp:|' sources/pyside2/cmake/Macros/PySideModules.cmake
+  - name: tor
+    buildsystem: simple
```
Why not use autotools here?

barthalion commented on org.onionshare.OnionShare.yaml:
```diff
@@ -0,0 +1,336 @@
+        url: https://dist.torproject.org/tor-0.4.5.6.tar.gz
+    modules:
+      - name: libevent
+        buildsystem: simple
```
Same as above.

barthalion commented on org.onionshare.OnionShare.yaml:
```diff
@@ -0,0 +1,336 @@
+        GOBIN: "/app/bin/"
+        GO111MODULE: "off"
+    build-commands:
+      - ". /usr/lib/sdk/golang/enable.sh; GOPATH=$PWD go install gitlab.com/yawning/obfs4.git/obfs4proxy"
```
Does it actually work? Flathub runs builds offline, https://github.com/flatpak/flatpak-builder-tools/tree/master/go-get can be used to generate manifest automatically.

barthalion commented on org.onionshare.OnionShare.yaml:
```diff
@@ -0,0 +1,336 @@
+      - type: dir
+        path: ../desktop/src
+    modules:
+      - name: python3-qrcode
```
There's a generator for pip too, if it makes the work simpler for you: https://github.com/flatpak/flatpak-builder-tools/tree/master/pip

barthalion commented:
bot, build org.onionshare.OnionShare

Erick555 commented on org.onionshare.OnionShare.yaml:
```diff
@@ -0,0 +1,336 @@
+  - "--talk-name=org.freedesktop.Flatpak"
+  - "--talk-name=org.freedesktop.Notifications"
+  - "--talk-name=org.freedesktop.secrets"
+  - "--filesystem=home:ro"
```
Can this app use portals? If yes then you may not need this permission.

micahflee commented:
bot, build org.onionshare.OnionShare

micahflee commented:
Okay, I've fixed the manifest and addressed all of your comments, and it seems ready to go.

> Your application has write access to `$XDG_CONFIG_HOME` by default.

Without this in `finish-args`:

```
   - "--filesystem=xdg-config/onionshare:create" 
```

I get this error:

```
$ flatpak run org.onionshare.OnionShare
OnionShare 2.3.1 | https://onionshare.org/
Gtk-Message: 19:29:10.970: Failed to load module "canberra-gtk-module"
Gtk-Message: 19:29:10.971: Failed to load module "canberra-gtk-module"
Qt: Session management error: None of the authentication protocols specified are supported
Traceback (most recent call last):
  File "/app/bin/onionshare", line 33, in <module>
  File "/app/lib/python3.8/site-packages/onionshare-2.3.1-py3.8.egg/onionshare/__init__.py", line 177, in main
OSError: [Errno 30] Read-only file system: '/home/user/.config/onionshare/lock'
```

The problem is in upstream OnionShare is writing [directly](https://github.com/micahflee/onionshare/blob/dfaa55d2a40edfb892e123f370f5d22b79e33d04/cli/onionshare_cli/common.py#L141) to `~/.config/onionshare` instead of to `$XDG_CONFIG_HOME` . So I'm keeping it for now, but I opened in issue upstream: https://github.com/micahflee/onionshare/issues/1293

I made the `tor` and `libevent` modules use autotools.

I actually did use the flatpak-builder-tools repo for both pip dependencies and for the golang obfs4proxy module, though it turns out https://github.com/flatpak/flatpak-builder-tools/tree/master/go-get doesn't work anymore as of about 6 days ago when https://github.com/flathub/org.freedesktop.Sdk.Extension.golang updated golang to 1.16 -- it took awhile to realize I needed the `GO111MODULE=off` environment variable for this older way of getting go sources to work.

micahflee commented:
bot, build org.onionshare.OnionShare

micahflee commented:
Oh yeah and

> Can this app use portals? If yes then you may not need this permission.

I'd like to use portals, but it's not yet implemented.

barthalion commented:
/merge
